service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -314,115 +196,82 @@ internal class SøknadsbehandlingServiceIverksettTest {
                     generertDokumentJson = "",
                 ).right()
             },
-        ).let {
-            it.søknadsbehandlingService.iverksett(
-                SøknadsbehandlingService.IverksettRequest(
-                    behandlingId = avslagTilAttestering.id,
-                    attestering = attestering,
-                ),
-            ) shouldBe expectedAvslag.right()
-
-            verify(it.søknadsbehandlingRepo).hent(avslagTilAttestering.id)
-            verify(it.brevService).lagDokument(argThat { it shouldBe beOfType<Avslagsvedtak.AvslagBeregning>() })
-            verify(it.søknadsbehandlingRepo).lagre(eq(expectedAvslag), anyOrNull())
-            verify(it.vedtakRepo).lagre(argThat { it is Avslagsvedtak.AvslagBeregning }, anyOrNull())
-            verify(it.brevService).lagreDokument(
-                argThat {
-                    it.metadata.sakId shouldBe avslagTilAttestering.sakId
-                    it.metadata.vedtakId shouldNotBe null
-                    it.metadata.bestillBrev shouldBe true
-                },
-                anyOrNull()
-            )
-            verify(it.behandlingMetrics).incrementAvslåttCounter(BehandlingMetrics.AvslåttHandlinger.PERSISTERT)
-            verify(it.ferdigstillVedtakService).lukkOppgaveMedBruker(any())
-            verify(it.observer).handle(
-                argThat {
-                    it shouldBe Event.Statistikk.SøknadsbehandlingStatistikk.SøknadsbehandlingIverksatt(
-                        expectedAvslag,
-                    )
-                },
-            )
-            it.verifyNoMoreInteractions()
-        }
+        )
+
+        serviceAndMocks.søknadsbehandlingService.iverksett(
+            SøknadsbehandlingService.IverksettRequest(
+                behandlingId = avslagTilAttestering.id,
+                attestering = attestering,
+            ),
+        ) shouldBe expectedAvslag.right()
+
+        verify(serviceAndMocks.søknadsbehandlingRepo).hent(avslagTilAttestering.id)
+        verify(serviceAndMocks.brevService).lagDokument(argThat { it shouldBe beOfType<Avslagsvedtak.AvslagBeregning>() })
+        verify(serviceAndMocks.søknadsbehandlingRepo).lagre(eq(expectedAvslag), anyOrNull())
+        verify(serviceAndMocks.vedtakRepo).lagre(argThat { it is Avslagsvedtak.AvslagBeregning }, anyOrNull())
+        verify(serviceAndMocks.brevService).lagreDokument(
+            argThat {
+                it.metadata.sakId shouldBe avslagTilAttestering.sakId
+                it.metadata.vedtakId shouldNotBe null
+                it.metadata.bestillBrev shouldBe true
+            },
+            anyOrNull()
+        )
+        verify(serviceAndMocks.behandlingMetrics).incrementAvslåttCounter(BehandlingMetrics.AvslåttHandlinger.PERSISTERT)
```

Comment:
Vi kan se på å fjerne disse gamle metrikkene etter denne PRen. Men hvis metrikken uansett skal være der, så er jeg usikker på om jeg tenker at vi burde teste det. Ihvertfall ikke her i den o-store-viktige-iverksettingstesten, kan isåfall heller være egne tester et sted på at ditten og datten fører til at vi pusher en metrikk.

Reply:
Problemet er at vi har `verifyNoMoreInteractions` på slutten, og da krasjer testen om vi ikke forteller om _alle_ service-kall